### PR TITLE
Update gem to support the latest Solidus version(4.4)

### PR DESCRIPTION
**Pull request by shahmayur001**
## Summary

This pull request resolves compatibility issues with Solidus version updates by adding necessary gem dependencies, modifying forms for proper field rendering, fixing jQuery issues with subscription creation, and updating permission sets. These changes align Solidus Subscriptions with the latest Solidus architecture. 

## Enhancements:
- **Form Rendering Restriction**  
  - Updated the subscription fields form to render only when the associated product is subscribable.

- **jQuery Script Fix**  
  - Corrected the script used for creating subscription line items to ensure the correct `subscribable_id` is stored.

- **Permission Set Updates**  
  - Introduced new privilege and category methods for the permission sets to maintain compatibility with the latest Solidus versions.

- **Fixed breadcrumbs issue**  
  - https://github.com/solidusio/solidus_subscriptions/issues/304
  
- **Wrap deface overrides in a module**  
  - Zeitwerk expects a class or module correctly named in these files in newer versions of Solidus using Zeitwrek.
  
  
## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

**Comment by fthobe** [comment]
@jarednorman we still struggle with SQLite testing, everything else looks ok. I think the test routine for SQLite is a little flawed. I have the impression it's more SQLite than us.

**Review by jarednorman — COMMENTED** [review]
We can merge without those passing until someone can fix them, but for future contributions, please split up the individual changes and explain them in the commit messages. The PR description is for additional information that's relevant to reviewers. The commits are the authoritative record of information about changes to the project.

**Comment by fthobe** [comment]
@jarednorman you are getting soft

**Comment by fthobe** [comment]
#305 cross reference

**Review by jarednorman — CHANGES_REQUESTED** [review]

File: app/overrides/views/admin_subscribable_product_checkbox.rb [hunk 1]
```diff
@@ -1,8 +1,12 @@
 # frozen_string_literal: true
 
-Deface::Override.new(
-  virtual_path: "spree/admin/products/_form",
-  name: "solidus_subscriptions_product_subscribable_checkbox",
-  insert_after: "[data-hook='admin_product_form_promotionable']",
-  partial: "spree/admin/products/subscribable_checkbox"
-)
+module Views
```
[the remark this answers is not in the document]
Comment (fthobe) [reply app/overrides/views/admin_subscribable_product_checkbox.rb:3]: Seems doable, mayur is implementing that today.

File: lib/solidus_subscriptions/subscription_line_item_builder.rb [hunk 2]
```diff
@@ -9,8 +9,6 @@ def create_subscription_line_item(line_item)
         subscription_params.merge(spree_line_item: line_item)
       )
 
-      # Rerun the promotion handler to pickup subscription promotions
-      ::Spree::PromotionHandler::Cart.new(line_item.order).activate
```
Comment (jarednorman) [inline lib/solidus_subscriptions/subscription_line_item_builder.rb:13]: I guess my only concern is this. Is this safe? Do we need to do something _different_ to handle legacy versus new promotions engines?

Comment (fthobe) [reply lib/solidus_subscriptions/subscription_line_item_builder.rb:13]: @jarednorman Man, like really, from the bottom of my heart, could you be more specific, I am sure when you write this kind of comments you already have a very specific way in mind to do it.

Comment (fthobe) [reply lib/solidus_subscriptions/subscription_line_item_builder.rb:13]: @jarednorman that is actually a tangible point and helps much more than the previous comment!!!

Comment (fthobe) [reply lib/solidus_subscriptions/subscription_line_item_builder.rb:13]: @jarednorman hey, sorry for replying so late. The issue of the removal of `::Spree::PromotionHandler::Cart.new(line_item.order).activate` was discussed here: https://github.com/gms-electronics/solidus_subscriptions/pull/1 If you have something else to consider please tell us.

**Review by jarednorman — APPROVED** [review]

**Pull request merged**